### PR TITLE
removes excessive runs from pod.bats

### PR DESCRIPTION
**Pull request by wgahnagl**
#### What type of PR is this?

> /kind api-change
> /kind bug

/kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
simplifies the pod.bats file by removing excessive use of run 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/pod.bats [hunk 1]
```diff
@@ -261,35 +184,17 @@ function teardown() {
...
-	run crictl stop "$ctr_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
-	run crictl stopp "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_config.json "$TESTDATA"/sandbox_config.json)
+	crictl start "$ctr_id"
+    crictl stop "$ctr_id"
```
Comment (umohnani8) [inline test/pod.bats:190]: nit: indentation

File: test/pod.bats [hunk 2]
```diff
@@ -13,82 +13,36 @@ function teardown() {
...
-	echo "$output"
-	[ "$status" -eq 0 ]
-	run crictl stopp "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
-	run crictl rmp "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+    ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_redis.json "$TESTDATA"/sandbox_config.json)
```
Comment (umohnani8) [inline test/pod.bats:27]: nit: indentation

File: test/pod.bats [hunk 3]
```diff
@@ -13,82 +13,36 @@ function teardown() {
...
-	[ "$status" -eq 0 ]
-	id="$output"
-	run crictl stopp "$id"
-	echo "$output"
-	[ "$status" -eq 0 ]
-	run crictl rmp "$id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	crictl stopp "$id"
```
Comment (kolyshkin) [inline test/pod.bats:17]: Cleanup is not needed as it's done in teardown. Should be removed in a separate patch, similar to e.g. https://github.com/cri-o/cri-o/pull/4253/commits/601e1e4cac7b930c1f1e42f8720d8cbdf9014c52

File: test/pod.bats (hunk 2 above)
Comment (kolyshkin) [inline test/pod.bats:27]: whitespace issue?

File: test/pod.bats (hunk 3 above)
Comment (kolyshkin) [reply test/pod.bats:17]: Unless, of course, it's part of the test, like in "pod remove".

Comment (wgahnagl) [reply test/pod.bats:17]: I think it is part of the test, since the test is supposed to check if the pod name is released when it's removed?

Comment (wgahnagl) [reply test/pod.bats:17]: think it is part of the test, since the test is supposed to check if the pod name is released when it's removed

File: test/pod.bats (hunk 2 above)
Comment (wgahnagl) [reply test/pod.bats]: should be fixed

Comment (wgahnagl) [reply test/pod.bats]: should be fixed

File: test/pod.bats (hunk 1 above)
Comment (wgahnagl) [reply test/pod.bats]: fixed

File: test/pod.bats (hunk 3 above)
Comment (kolyshkin) [reply test/pod.bats:17]: I meant the second cleanup at the end of the function (lines 20-21 now).

File: test/pod.bats [hunk 4]
```diff
@@ -261,35 +184,17 @@ function teardown() {
...
-	echo "$output"
-	[ "$status" -eq 0 ]
-	run crictl stopp "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_config.json "$TESTDATA"/sandbox_config.json)
+	crictl start "$ctr_id"
+	crictl stop "$ctr_id"
+    crictl stopp "$pod_id"
```
Comment (kolyshkin) [inline test/pod.bats:191]: nit: whitespace

File: test/pod.bats [hunk 5]
```diff
@@ -343,19 +242,10 @@ function teardown() {
...
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_id="$output"
-
-	run crictl create "$pod_id" "$TESTDATA"/container_config_sleep.json "$TESTDIR"/sandbox_config-systemd.json
-	echo "$output"
-	[ "$status" -eq 0 ]
-	ctr_id="$output"
+	pod_id=$(crictl runp "$TESTDIR"/sandbox_config-systemd.json)
+	ctr_id=$( crictl create "$pod_id" "$TESTDATA"/container_config_sleep.json "$TESTDIR"/sandbox_config-systemd.json)
```
Comment (kolyshkin) [inline test/pod.bats:246]: nit: whitespace
```suggestion
	ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_config_sleep.json "$TESTDIR"/sandbox_config-systemd.json)
```

File: test/pod.bats (hunk 3 above)
Comment (wgahnagl) [reply test/pod.bats:17]: oh, I see what you mean

File: test/pod.bats (hunk 5 above)
Comment (wgahnagl) [reply test/pod.bats]: should be fixed !

File: test/pod.bats (hunk 4 above)
Comment (wgahnagl) [reply test/pod.bats]: should be fixed !

**Comment by kolyshkin** [comment]
1. couple of whitespace nits.
2. in some cases there's an extra cleanup which should not be there.

For example, `@test "pod release name on remove"` should look like this (schematically):

```
runp
stopp
rmp
runp
```

i.e. the second cleanup should not be part of the test as we're checking the pod with the same name can be recreated, and not checking that the second pod can be removed.

**Comment by kolyshkin** [comment]
1. Looks like "pod list filtering" test has unneeded cleanup as well.

2. As it comes for commit messages, it would be nice to have subjects like
```
test/pod.bats: ...
```
so a reader would see where the changes are made right away. As someone who reads git history a lot, I can attest that if the message starts with a file name, directory, subsystem name, package name, top-level feature or something like this, it makes it easier to visually filter what you're looking for.

Of course this is super minor. I wish I can point out to some kind of "best practices" doc for git commits but I don't have any :(

File: test/pod.bats [hunk 6]
```diff
@@ -343,19 +242,10 @@ function teardown() {
...
-	ctr_id="$output"
+	pod_id=$(crictl runp "$TESTDIR"/sandbox_config-systemd.json)
+	ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_config_sleep.json "$TESTDIR"/sandbox_config-systemd.json)
 
-	run crictl start "$ctr_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	crictl start "$ctr_id"
 
 	run systemctl show "crio-${ctr_id}.scope"
```
Comment (kolyshkin) [inline test/pod.bats:242]: Here, as well, since 0 exit code is expected, you can do `output=$(systemctl show "crio-${ctr_id}.scope")`. Sorry I missed it earlier.

File: test/pod.bats [hunk 7]
```diff
@@ -368,26 +258,14 @@ function teardown() {
...
-
-
-	run crictl inspectp "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	crictl inspectp "$pod_id"
 
 	conf_pause_image=$(grep -oP 'pause_image = \K"[^"]+"' "$CRIO_CONFIG")
 	pod_pause_image=$(echo "$output" | jq -e .info.image)
```
Comment (kolyshkin) [inline test/pod.bats:257]: With your changes above, this line won't work. You probably need to do
```bash
output=$(crictl inspectp "$pod_id")
```

File: test/pod.bats [hunk 8]
```diff
@@ -196,104 +107,54 @@ function teardown() {
...
+	crictl stopp "$pod_id"
 }
 
 @test "pod remove idempotent" {
 	start_crio
-	run crictl runp "$TESTDATA"/sandbox_config.json
-	echo "$output"
-	[ "$status" -eq 0 ]
+	output=$(crictl runp "$TESTDATA"/sandbox_config.json)
 	pod_id="$output"
```
Comment (kolyshkin) [inline test/pod.bats:130]:
```suggestion
	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
```

File: test/pod.bats [hunk 9]
```diff
@@ -196,104 +107,54 @@ function teardown() {
...
-	run crictl stopp "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
-	run crictl rmp "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	output=$(crictl stopp "$pod_id")
+	output=$(crictl rmp "$pod_id")
```
Comment (kolyshkin) [inline test/pod.bats:132]:
```suggestion
	crictl stopp "$pod_id"
	crictl rmp "$pod_id"
```

File: test/pod.bats [hunk 10]
```diff
@@ -324,14 +183,8 @@ function teardown() {
...
 	CONTAINER_DROP_INFRA_CTR=false start_crio
-	run crictl runp "$TESTDIR"/sandbox_systemd_cgroup_parent.json
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_id="$output"
-
-	run systemctl list-units --type=slice
-	echo "$output"
-	[ "$status" -eq 0 ]
+	pod_id=$(crictl runp "$TESTDIR"/sandbox_systemd_cgroup_parent.json)
```
Comment (kolyshkin) [inline test/pod.bats:178]: Looks like `pod_id` is not used, so this assignment is useless.

```suggestion
	crictl runp "$TESTDIR"/sandbox_systemd_cgroup_parent.json
```

**Comment by kolyshkin** [comment]
Left a couple of comments.

Also, this needs to be rebased.

Ideally, I'd like to have the "remove unnecessary cleanup" commit first, then the "excessive run removal" one. This way the commits will be smaller (and easier to review!).

File: test/pod.bats [hunk 11]
```diff
@@ -299,14 +174,8 @@ function teardown() {
...
 	CONTAINER_DROP_INFRA_CTR=false start_crio
-	run crictl runp "$TESTDIR"/sandbox_systemd_cgroup_parent.json
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_id="$output"
-
-	run systemctl list-units --type=slice
-	echo "$output"
-	[ "$status" -eq 0 ]
+    crictl runp "$TESTDIR"/sandbox_systemd_cgroup_parent.json
```
Comment (kolyshkin) [inline test/pod.bats:177]: nit: whitespace

File: test/pod.bats [hunk 12]
```diff
@@ -318,39 +187,23 @@ function teardown() {
...
-	[ "$status" -eq 0 ]
+	crictl start "$ctr_id"
 
-	run systemctl show "crio-${ctr_id}.scope"
-	[ "$status" -eq 0 ]
-	echo "$output" | grep 'TimeoutStopUSec=' || true      # show
-	echo "$output" | grep -q '^TimeoutStopUSec=1min 28s$' # check
+	output=$(systemctl show "crio-${ctr_id}.scope")
+	echo "$output" | grep 'TimeoutStopUSec=' || true	# show
+	echo "$output" | grep -q '^TimeoutStopUSec=1min 28s$'	# check
```
Comment (kolyshkin) [inline test/pod.bats:197]: Hmm, can't understand why these two lines have changed

**Review by kolyshkin — CHANGES_REQUESTED** [review]

**Review by kolyshkin — APPROVED** [review]
LGTM

**Review by mrunalp — APPROVED** [review]

**Comment by mrunalp** [comment]
@wgahnagl could you edit and add release note?
